visualization/app/codeCharta/ui/codeMap/codeMap.label.service.ts

```diff
@@ -62,51 +62,49 @@ export class CodeMapLabelService implements CameraChangeSubscriber {
 		const labelYOrigin = y + node.height
 		const labelZ = (z + node.length / 2) * multiplier.z
 
-		if (node.attributes?.[state.dynamicSettings.heightMetric]) {
-			let labelText = ""
+		let labelText = ""
 
-			if (options.showNodeName) {
-				labelText = `${node.name}`
-			}
-			if (options.showNodeMetric) {
-				if (labelText !== "") {
-					labelText += "\n"
-				}
-				labelText += `${node.attributes[state.dynamicSettings.heightMetric]} ${state.dynamicSettings.heightMetric}`
+		if (options.showNodeName) {
+			labelText = `${node.name}`
+		}
+		if (options.showNodeMetric) {
+			if (labelText !== "") {
+				labelText += "\n"
 			}
+			labelText += `${node.attributes[state.dynamicSettings.heightMetric]} ${state.dynamicSettings.heightMetric}`
+		}
 
-			const label = this.makeText(labelText, 30, node)
-			const {
-				appSettings: { layoutAlgorithm }
-			} = state
-			const labelHeightScaled = this.LABEL_HEIGHT_COEFFICIENT * margin * this.LABEL_SCALE_FACTOR
-			let labelOffset = labelHeightScaled + label.heightValue / 2
-
-			switch (layoutAlgorithm) {
-				// !remark : algorithm scaling is not same as the squarified layout,
-				// !layout offset needs to be scaled down,the divided by value is just empirical,
-				// TODO !needs further investigation
-				case LayoutAlgorithm.StreetMap:
-				case LayoutAlgorithm.TreeMapStreet:
-					labelOffset /= 10
-					this.LABEL_HEIGHT_POSITION = 0
-					label.line = this.makeLine(labelX, labelY + labelOffset, labelYOrigin, labelZ)
-					break
-				default:
-					label.line = this.makeLine(labelX, labelY + labelHeightScaled / 2, labelYOrigin, labelZ)
-			}
+		const label = this.makeText(labelText, 30, node)
+		const {
+			appSettings: { layoutAlgorithm }
+		} = state
+		const labelHeightScaled = this.LABEL_HEIGHT_COEFFICIENT * margin * this.LABEL_SCALE_FACTOR
+		let labelOffset = labelHeightScaled + label.heightValue / 2
+
+		switch (layoutAlgorithm) {
+			// !remark : algorithm scaling is not same as the squarified layout,
+			// !layout offset needs to be scaled down,the divided by value is just empirical,
+			// TODO !needs further investigation
```

Comment:
We might find a nicer description here but I can't think of anything right now :)